### PR TITLE
GPU ppm type 0 (GPU hydro PR 3/n)

### PR DESCRIPTION
**Pull request by harpolea**
This offloads `slope.F90` to the GPU (used by ppm type 0). It has been tested for the 2d and 3d reacting bubble problems compiled for MPI and GPU. 

Note: It is likely that this version will not work if compiled with OpenMP (this will come in a future PR)

### CODE REVIEW

**Comment by zingale** [comment]
all tests pass with this PR

File: Source/Src_nd/compute_dt.F90 [hunk 1]
```diff
@@ -472,14 +475,14 @@ subroutine firstdt(lev, dt, umax, lo, hi, dx, &
        else
           dt_sound = cfl / max(spdx,spdy,spdz)
        end if
-       call amrex_min(dt,dt_sound)
+       dt = min(dt,dt_sound)
     end if
```
Comment (zingale) [inline Source/Src_nd/compute_dt.F90]: don't we need `amrex_min` still?

Comment (harpolea) [reply Source/Src_nd/compute_dt.F90]: I remember when I was writing this there were some changes being made to the dt routines, so I probably ended up merging the wrong thing at some point. There shouldn't be any changes in the dt stuff compared to development, so I'll fix this

**Comment by zingale** [comment]
I'll run the suite on this one more time and then merge

**Pull request merged**